amazonclouddrive/amazonclouddrive.go

```diff
@@ -668,6 +668,51 @@ func (f *Fs) Hashes() fs.HashSet {
 	return fs.HashSet(fs.HashMD5)
 }
 
+func (f *Fs) Move(src fs.Object, remote string) (fs.Object, error) {
+	srcObj, ok := src.(*Object)
+	if !ok {
+		fs.Debug(src, "Can't move - not same remote type")
+		return nil, fs.ErrorCantMove
+	}
+
+	// Temporary Object under construction
+	dstObj := &Object{
+		fs:     f,
+		remote: remote,
+	}
+
+	var err error
+	_, directoryID, err := f.dirCache.FindPath(remote, false)
+	if err != nil {
+		return nil, err
+	}
+
+	var info *acd.Node
+	var resp *http.Response
+	if directoryID == srcObj.info.Parents[0] {
+		// Do the rename
+		err = f.pacer.Call(func() (bool, error) {
+			info, resp, err = srcObj.info.Rename(remote)
+			return srcObj.fs.shouldRetry(resp, err)
+		})
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Do the move
+		err = f.pacer.Call(func() (bool, error) {
+			info, resp, err = srcObj.info.Move(directoryID)
```

Comment:
You're right there actually, it needs some more changes made before it's ready to commit. It did work for my test cases, I didn't really think more broadly about it. Which is my mistake sorry.

I'm still wrapping my head around some of the structure of rclone, it should be possible to rename and move in the same call. If you move it then rename it. Or perhaps the other way around, it's quite annoying you can't do both in the same call with ACD's API.

Perhaps comparing the leafs with an if statement after move, and if they aren't equal also applying a rename, although in the case would there need to be a check in here to prevent clashing? Or is that handled elsewhere
Such that:
a is in the root
b is in newfolder
You then:
mv a newfolder/b
Parent is changed ok
Then the rename operation happens and fails due to b already existing.

I'll see if I can find some time to correct this, however if you want to just take what I have here and complete it sooner that might be better since I'm still learning the inner logic of rclone :)